subsystem/src/subsystem.rs

```diff
@@ -108,58 +116,82 @@ impl<T> Clone for Handle<T> {
 
 #[derive(Debug, Ord, PartialOrd, PartialEq, Eq, Clone, Copy, thiserror::Error)]
 pub enum CallError {
-    #[error("Callee subsystem has terminated")]
-    SubsystemDead,
+    #[error("Call submission failed")]
+    SubmissionFailed,
+    #[error("Result retrieval failed")]
+    ResultFetchFailed,
+}
+
+pub struct CallResult<T>(Result<CallResponse<T>, CallError>);
+
+impl<T> CallResult<T> {
+    /// Get the corresponding [`CallResponse`], with the opportunity to handle errors at send time.
+    pub fn response(self) -> Result<CallResponse<T>, CallError> {
```

Comment:
Maybe adding a test with this usage as example will help. The only example I see is in github.

Reply:
Added